File: windows/Scanner/Detectors/WindsurfDetector.cs
using AIExposureScanner.Scanner.Filesystem;
using AIExposureScanner.Scanner.Models;
using AIExposureScanner.Scanner.Parsing;

namespace AIExposureScanner.Scanner.Detectors;

public sealed class WindsurfDetector : IDetector
{
    public string Id => "windsurf";
    public string DisplayName => "Windsurf";

    public ScanFacts CollectFacts(IFilesystemFacade fs)
    {
        var mcpPath = DetectorSupport.HomePath(fs, ".codeium/windsurf/mcp_config.json");
        var settingsPath = Path.Combine(fs.ApplicationDataDirectory, "Windsurf", "User", "settings.json");
        var facts = new ScanFacts();
        DetectorSupport.AppendAppInstallationFact(
            facts,
            fs,
            Id,
            [Path.Combine(fs.LocalApplicationDataDirectory, "Programs", "Windsurf")]
        );

        var text = DetectorSupport.ReadIfPresent(fs, mcpPath);
        if (text is not null)
        {
            facts.McpServers.AddRange(FactParsers.ParseMcpServersJson(text, Id, mcpPath));
            DetectorSupport.AppendConfigFileFact(facts, fs, Id, mcpPath);
        }

        DetectorSupport.AppendConfigFileFact(facts, fs, Id, settingsPath);
        return facts;
    }
}

File: macos/Scanner/Sources/Scanner/Detectors/WindsurfDetector.swift
public struct WindsurfDetector: Detector {
    public let id = "windsurf"
    public let displayName = "Windsurf"

    public init() {}

    public func collectFacts(fs: any FilesystemFacade) throws -> ScanFacts {
        let mcpPath = homePath(fs, ".codeium/windsurf/mcp_config.json")
        let settingsPath = homePath(fs, "Library/Application Support/Windsurf/User/settings.json")
        var facts = ScanFacts()
        appendAppInstallationFact(
            &facts,
            fs: fs,
            appId: id,
            candidatePaths: ["/Applications/Windsurf.app", homePath(fs, "Applications/Windsurf.app")]
        )

        if let text = try readIfPresent(fs, path: mcpPath) {
            facts.mcpServers.append(contentsOf: try FactParsers.parseMcpServersJson(text, appId: id, configPath: mcpPath))
            appendConfigFileFact(&facts, fs: fs, appId: id, path: mcpPath)
        }

        appendConfigFileFact(&facts, fs: fs, appId: id, path: settingsPath)
        return facts
    }
}
